### PR TITLE
Support Erlang 24

### PR DESCRIPTION
**Pull request by x1001100011**
Support Erlang 24:

1. add both erlang23/24 to ci
2. upgrade rebar3 and getopt version to fix system:get_stacktrace issue
3. upgrade cowlib and meck to fix warning
4. depends on the latest emqx repo master branch

### CODE REVIEW

**Review by k32 — APPROVED** [review]

File: rebar.config [hunk 1]
```diff
@@ -22,8 +22,8 @@
 {profiles,
     [{test,
         [{deps,
-            [{meck, "0.8.13"},
-             {emqx, {git, "https://github.com/emqx/emqx", {branch, "master"}}},
+            [{meck, "0.9.2"},
+             {emqx, {git_subdir, "https://github.com/emqx/emqx", {branch, "master"}, "apps/emqx"}},
```
Comment (zmstone) [inline rebar.config:26]: awesome

Comment (qzhuyan) [reply rebar.config:26]: so this is the emqx's rebar3 feature?

Comment (x1001100011) [reply rebar.config:26]: @qzhuyan it's an official feature http://rebar3.org/docs/configuration/dependencies/

File: test/emqtt_quic_SUITE.erl [hunk 2]
```diff
@@ -49,7 +49,10 @@ t_quic_sock(Config) ->
     quic_server:stop(Server).
 
 send_and_recv_with(Sock) ->
-    {ok, {{_,_,_,_}, _}} = emqtt_quic:sockname(Sock),
+    case emqtt_quic:sockname(Sock) of
```
Comment (zmstone) [inline test/emqtt_quic_SUITE.erl:52]: use tuple_size to validate

**Review by zmstone — APPROVED** [review]

**Pull request merged**